core/gather/driver/prepare.js

```diff
@@ -29,7 +31,13 @@ async function enableAsyncStacks(session) {
   // See https://bugs.chromium.org/p/chromium/issues/detail?id=990945&q=setSkipAllPauses&can=2
   session.on('Page.frameNavigated', event => {
     if (event.frame.parentId) return;
-    enable().catch(err => log.error('Driver', err));
+    enable().catch(err => {
```

Comment:
mentioned this in chat, but maybe it's better to move this to the instrumentation or sensitiveInstrumentation periods and purposefully disable it after it's done? Then the driver is known connected (unless there's a major error) and keeps this as-is

Reply:
https://github.com/GoogleChrome/lighthouse/pull/14717 is one way to do that, although I think your suggesting doing this outside a gatherer right?

Reply:
Yes, looking at the balance between gatherers taking care of themselves and gatherers actually being independent units that don't interfere with each other, can be ordered however, etc. It's definitely helpful to have something as impactful as having the `Debugger` domain turned on out here where it can be turned on and off independent of the particular gatherer, and other future gatherers wouldn't have to coordinate (if we ever do style invalidation or whatever).

Reply:
I agree with your concerns, however I think they would be better solved by creating a separate session for the the async stack stuff. That would allow us to encapsulate the async stack stuff in the DevtoolsLog gatherer *and* keep it's handling of the debugger domain independent from other instrumentation.